embassy-nxp/src/usart/lpc55.rs

```diff
@@ -0,0 +1,854 @@
+        while !buffer.is_empty() {
+        match Self::drain_fifo(self, buffer) {
+                Ok(0) => continue, // Wait for more data
+                Ok(n) => buffer = &mut buffer[n..],
+                Err((_, err)) => return Err(err),
+            }
+        }
+        Ok(())    
+    }
+
+    fn drain_fifo(&mut self, buffer: &mut [u8]) -> Result<usize, (usize, Error)> {
+        T::drain_fifo(buffer)
+    }
+}
+
+impl<'d, T: Instance> UsartRx<'d, T, Blocking> {
+    pub fn new_blocking(_usart: Peri<'d, T>, rx: Peri<'d, impl RxPin<T>>, config: Config) -> Self {
+        Usart::<T, Blocking>::init(None, Some(rx.into()), config);
+        Self::new_inner()
+    }
+}
+
+impl<'d, T: Instance> Usart<'d, T, Blocking> {
+    pub fn new_blocking(
+        usart: Peri<'d, T>,
+        tx: Peri<'d, impl TxPin<T>>,
+        rx: Peri<'d, impl RxPin<T>>,
+        config: Config,
+    ) -> Self {
+        Self::new_inner(usart, tx.into(), rx.into(), config)
+    }
+}
+
+impl<'d, T: Instance, M: Mode> Usart<'d, T, M> {
+    fn new_inner(_usart: Peri<'d, T>, mut tx: Peri<'d, AnyPin>, mut rx: Peri<'d, AnyPin>, config: Config) -> Self {
+        Self::init(Some(tx.reborrow()), Some(rx.reborrow()), config);
+        Self {
+            tx: UsartTx::new_inner(),
+            rx: UsartRx::new_inner(),
+        }
+    }
+
+    fn init(_tx: Option<Peri<'_, AnyPin>>, _rx: Option<Peri<'_, AnyPin>>, config: Config) {
+        T::enable_clock();
+        T::reset_flexcomm();
+        
+        let source_clock: u32 = T::select_clock(config.baudrate);
+        T::configure_flexcomm();
+        T::tx_pin_config();
+        T::rx_pin_config();
+        Self::set_baudrate(source_clock, config.baudrate);
+        T::configure_usart(config);
+        T::configure_dma();
+        T::enable_usart();       
+    }
+
+
+    fn set_baudrate(source_clock: u32, baudrate: u32) {
+        let brg_value = source_clock / (16 * baudrate); 
+        let raw_clock = source_clock / (16 * brg_value);
+        let mult_value: u32 = (raw_clock * 256 / baudrate) - 256;
+    
+        assert!(mult_value < 256);
+        assert!(brg_value < 256);
+        
+        T::set_baudrate(mult_value as u8, brg_value as u8);   
+    }
+}
+
+impl<'d, T: Instance, M: Mode> Usart<'d, T, M> {
+    /// Transmit the provided buffer blocking execution until done.
+    pub fn blocking_write(&mut self, buffer: &[u8]) -> Result<(), Error> {
+        self.tx.blocking_write(buffer)
+    }
+
+    /// Flush USART TX blocking execution until done.
+    pub fn blocking_flush(&mut self) -> Result<(), Error> {
+        self.tx.blocking_flush()
+    }
+
+    /// Read from USART RX blocking execution until done.
+    pub fn blocking_read(&mut self, buffer: &mut [u8]) -> Result<(), Error> {
+        self.rx.blocking_read(buffer)
+    }
+
+    /// Check if UART is busy transmitting.
+    pub fn busy(&self) -> bool {
+        self.tx.busy()
+    }
```

Comment:
nitpick. Although documented and generally obvious this refers to the `tx`, I could see a world where at a glance this could be interpreted as `RX or TX is currently being used by another task` when non-blocking USART gets made/implemented.

```suggestion
    /// Check if UART is busy transmitting.
    pub fn tx_busy(&self) -> bool {
        self.tx.busy()
    }
```